packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx

```diff
@@ -70,6 +70,20 @@ export function GridColumnsPanel(props: GridColumnsPanelProps) {
   const ownerState = { classes: rootProps.classes };
   const classes = useUtilityClasses(ownerState);
 
+  const sortedColumns = React.useMemo(() => {
+    if (rootProps.componentsProps?.columnsPanel.sort === 'asc') {
+      return [...columns].sort((a, b) =>
+        (a.headerName || a.field) > (b.headerName || b.field) ? 1 : -1,
```

Comment:
To let sorting adapt to different locales, you can use `Intl.Collator();`
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator

This behavior has already be done in the filter panel [here](https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx#L228:L243)